src/main/java/com/tripply/booking/service/Impl/RoomServiceImpl.java

```diff
@@ -0,0 +1,136 @@
+package com.tripply.booking.service.Impl;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import com.opencsv.CSVReader;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import com.tripply.booking.Exception.BadRequestException;
+import com.tripply.booking.Exception.DataNotFoundException;
+import com.tripply.booking.entity.Hotel;
+import com.tripply.booking.entity.Room;
+import com.tripply.booking.model.ResponseModel;
+import com.tripply.booking.model.response.RoomUploadResponse;
+import com.tripply.booking.repository.HotelRepository;
+import com.tripply.booking.repository.RoomRepository;
+import com.tripply.booking.service.RoomService;
+
+@Service
+public class RoomServiceImpl implements RoomService {
+	
+	@Autowired
+	private HotelRepository hotelRepository;
+
+	@Autowired
+	private RoomRepository roomRepository;
+
+	@Override
+	public ResponseModel<List<RoomUploadResponse>> uploadRooms(UUID hotelId, MultipartFile file) throws Exception {
+		Optional<Hotel> hotelOptional = hotelRepository.findById(hotelId);
+		if (!hotelOptional.isPresent()) {
+	         throw new DataNotFoundException("Hotel not Found");
+		}
+
+		Hotel hotel = hotelOptional.get();
+        List<Room> rooms = new ArrayList<>();
+        List<RoomUploadResponse> responses = new ArrayList<>();
+
+        String fileType = file.getContentType();
+        if (fileType.equals("text/csv")) {
+            rooms = parseCSV(file, responses);
+        } else if (fileType.equals("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet")) {
+            rooms = parseExcel(file, responses);
+        } else {
+            throw new BadRequestException("Invalid file format");
+        }
+        for (Room room : rooms) {
```

Comment:
Add here seperate method to add hotel rooms. Use async annotation on that method, as we are doing bulk upload and this process takes time

Reply:
Its data read from cell one by one in Excel and Csv file. there is synchronization process needed.I don't see async annotation on this method.

Reply:
Ok tell me. if csv have 1000 records , then how much time your API will take?

Reply:
okay I will R&D then Modify the code.